src/components/quests-list/index.js

```diff
@@ -68,12 +81,38 @@ const getQuestList = (questList, t) => {
 function QuestsList(props) {
     const { itemQuests } = props;
     const { t } = useTranslation();
+    const [showAllQuests, setShowAllQuests] = useState(false);
+    const settings = useSelector((state) => state.settings);
     let title = t('Quests Requiring');
     if (itemQuests.length > 0 && itemQuests[0].rewardItems) title = t('Quests Providing');
+    let toggleFilter = '';
+    if (settings.completedQuests?.length > 0) toggleFilter = (
+        <Filter>
+            <ToggleFilter
+                checked={showAllQuests}
+                label={t('Show completed')}
+                onChange={(e) =>
+                    setShowAllQuests(!showAllQuests)
+                }
+                tooltipContent={
+                    <div>
+                        {t(
+                            'Shows all quests regardless of if you\'ve completed them',
+                        )}
+                    </div>
+                }
+            />
+        </Filter>
+    );
     return (
         <div>
-            <h2>{title}</h2>
-            {getQuestList(itemQuests, t)}
+            <div className="item-quest-headline-wrapper">
+                <h2>
+                    {title}
+                </h2>
+                {toggleFilter}
```

Comment:
Why we don't want to show the toggle if there is no completed quests by show it regardless of the built module for hideout?

Reply:
First, the quests tables are a the inverse of the other tables. For crafts and barters, the more you progress, the more items will be shown in the tables. For quests, the more you progress, the more items are hidden in the tables. If the user hasn't completed any quests, then toggling those filters on will have no impact, and there's no reason to show them.

Second, there's currently no way to mark quests as complete aside from using the TarkovTracker sync. No quests complete likely means the user hasn't set up the TarkovTracker sync, so they'd be confused about where they're supposed to log their quest completion.

Reply:
I see your point and agree with it, still the same points apply to hideout modules:
* The first point is true for both quests and hideout modules.
* The second point is also half true for hideout modules as without TarkovTracker sync we can only know about the level of modules needed for crafts and there is a tonne of items needed for modules we can't know anything about that will have the switch shown anyway. 
But the fact that it's half-and-half I see why we can decide to leave the switch always shown for hideout modules.

But instead of quest completion, I think it will be better if we decide whether to show or not the switch if the TarkovTracker sync is enabled. 
This is the UI purist that speaks in me, but the UI must be consistent. Showing the switch on some items (if you have completed quests) and not showing it on others (if you didn't complete the quest) is actually more confusing

Reply:
> Showing the switch on some items (if you have completed quests) and not showing it on others (if you didn't complete the quest) is actually more confusing

But this isn't every how it works. The toggles on both quest tables are either both shown or both hidden. The toggle will never be shown on one of the quest tables and hidden on the other.

Showing or hiding the toggle based on whether TarkovTracker sync is is not a great solution. Someone could enable TarkovTracker sync which would then sync their current quest completion. If they then disable TarkovTracker sync, their quest completion isn't cleared, it just isn't updated anymore. But the Tarkov.dev website interface provides no way to modify the locally-stored quest completion. Based on the quests that are marked as complete from the stale TarkovTracker sync, the quest tables could hide items but then also hide the toggles because TarkovTracker sync is disabled, thus hiding items from the user with no way for the user to access the toggles.

When no quests are complete, the toggles do nothing. How is it bad UI to hide controls from the user that do nothing?

Reply:
>> Showing the switch on some items (if you have completed quests) and not showing it on others

> But this isn't how it works. The toggles on both quest tables are either both shown or both hidden. The toggle will never be shown on one of the quest tables and hidden on the other.

I meant on different items, not the same item. Like when you switch page to a different item and the switch is gone (or appears).

> If they then disable TarkovTracker sync, their quest completion isn't cleared

This is an issue, I didn't know that it was not cleared. If there is no way a user can clean the progression with tarkov.dev UI then we need to clean the progression ourselves when the user disables sync.

> When no quests are complete, the toggles do nothing.

This is the same for 'Ignore settings' in crafts when the max station level is built, the switch does nothing. 

> I don't really know the tenets of UI, but it doesn't seem like bad design to me to hide controls that would definitely have no effect.

Think about any app's menu that was touched by a UI/UX developer, if you can't use the menu/button then it is disabled, not hidden. Consistency in UI element placement and functionality is a big topic in UI/UX design. 

---

Anyway, this whole discussion is just my inner UI purist speaking. I'm no expert in the field, just a self-learn enthusiast 😅 so take this with a grain of salt and let's ship 😉 🚀